addons/website_sale/models/website.py

```diff
@@ -568,74 +567,77 @@ def _send_abandoned_cart_email(self):
-                    'back_button': str,
-                    'back_button_href': str
-                }
-            ]
-        """
+    def _display_partner_b2b_fields(self):
+        """ This method is to be inherited by localizations and return
+        True if localization should always displayed b2b fields """
         self.ensure_one()
-        is_extra_step_active = self.viewref('website_sale.extra_info').active
-        redirect_to_sign_in = self.account_on_checkout == 'mandatory' and self.is_public_user()
-
-        steps = [(['website_sale.cart'], {
-            'name': _lt("Review Order"),
-            'current_href': '/shop/cart',
-            'main_button': _lt("Sign In") if redirect_to_sign_in else _lt("Checkout"),
-            'main_button_href': f'{"/web/login?redirect=" if redirect_to_sign_in else ""}/shop/checkout?try_skip_step=true',
-            'back_button':  _lt("Continue shopping"),
-            'back_button_href': '/shop',
-        }), (['website_sale.checkout', 'website_sale.address'], {
-            'name': _lt("Delivery"),
-            'current_href': '/shop/checkout',
-            'main_button': _lt("Confirm"),
-            'main_button_href': f'{"/shop/extra_info" if is_extra_step_active else "/shop/confirm_order"}',
-            'back_button':  _lt("Back to cart"),
-            'back_button_href': '/shop/cart',
-        })]
-        if is_extra_step_active:
-            steps.append((['website_sale.extra_info'], {
-                'name': _lt("Extra Info"),
-                'current_href': '/shop/extra_info',
-                'main_button': _lt("Continue checkout"),
-                'main_button_href': '/shop/confirm_order',
-                'back_button':  _lt("Back to delivery"),
-                'back_button_href': '/shop/checkout',
-            }))
-        steps.append((['website_sale.payment'], {
-            'name': _lt("Payment"),
-            'current_href': '/shop/payment',
-            'back_button':  _lt("Back to delivery"),
-            'back_button_href': '/shop/checkout',
-        }))
+
+        return self.is_view_active('website_sale.address_b2b')
+
+    @api.model_create_multi
+    def create(self, vals_list):
+        websites = super().create(vals_list)
+        for website in websites:
+            website._create_checkout_steps()
+        return websites
+
+    def _create_checkout_steps(self):
+        generic_steps = self.env['website.checkout.step'].sudo().search([
+            ('website_id', '=', False),
+            ('active', '=', False),
+        ])
+        for step in generic_steps:
+            is_published = bool(step.step_href != '/shop/extra_info')
+            step.copy({'website_id': self.id, 'is_published': is_published, 'active': True})
+
+    def _get_website_specific_step(self, href):
+        return self.env['website.checkout.step'].sudo().search([
+            ('website_id', '=', self.id),
+            ('step_href', '=', href),
+        ], limit=1)
```

Comment:
The name doesn't really makes sense anymore as all checkout steps are website specific.